src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -69,11 +69,14 @@ public static int GetDirtyNodeCountAtGlobalScope(Executable exe)
             int dirtyNodes = 0;
             var graph = exe.instrStreamList[0].dependencyGraph;
             var graphNodes = graph.GetGraphNodesAtScope(Constants.kInvalidIndex, Constants.kGlobalScope);
-            foreach (AssociativeGraph.GraphNode graphNode in graphNodes)
+            if (graphNodes != null)
             {
-                if (graphNode.isDirty)
+                foreach (AssociativeGraph.GraphNode graphNode in graphNodes)
                 {
-                    ++dirtyNodes;
+                    if (graphNode.isDirty)
+                    {
+                        ++dirtyNodes;
+                    }
```

Comment:
Done, Ive replaced this with a simpler function that determines if at least one graphnode is dirty at the global scope.
The old function was implemetned for different uses but is now removed